File: javascript/math/231. Power of Two.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfTwo = function(n) {
    if (n < 1) return false;
    if (n == 1) return true;
    
    return isPowerOfTwo(n / 2);
};

var isPowerOfTwo = function(n) {
    return n > 0 && ((n & n-1) === 0);
};

File: python/math/231. Power of Two.py
'''
own: Recursion
Time: O(n)
Space: O(1)
'''

class Solution:
    # own
    def isPowerOfTwo(self, n: int) -> bool:
        if n == 1:
            return True
        if n < 1:
            return False

        return self.isPowerOfTwo(n / 2)
    

    def isPowerOfTwo(self, n: int) -> bool:
        return n > 0 and ((n & n-1) == 0)